### PR TITLE
Fix Query count error when call to webservice fails

### PR DESCRIPTION
**Pull request by alaxos**
Closes UseMuffin/Webservice#33

### CODE REVIEW

File: src/Query.php [hunk 1]
```diff
@@ -428,7 +428,11 @@ public function count()
             $this->_execute();
         }
 
-        return $this->__resultSet->total();
+        if(is_a($this->__resultSet, '\Cake\Datasource\ResultSetInterface')) {
+            return $this->__resultSet->total();
+        } else {
+            return false;
```
Comment (davidyell) [inline src/Query.php]: I would think this should return 0 and not false.

Comment (ADmad) [inline src/Query.php]: Yes, the `count()` method is supposed to return integer.

File: src/Query.php [hunk 2]
```diff
@@ -428,7 +428,11 @@ public function count()
             $this->_execute();
         }
 
-        return $this->__resultSet->total();
+        if(is_a($this->__resultSet, '\Cake\Datasource\ResultSetInterface')) {
```
Comment (ADmad) [inline src/Query.php]: Space needed after `if`. Use `instanceof` instead of `is_a()`.

**Comment by ADmad** [comment]
@alaxos Ping

File: src/Query.php (hunk 1 above)
Comment (alaxos) [inline src/Query.php]: Returning 0 if something wrong happens during the query execution would let think it really did return 0 items. Is it something that we really want ? If the `execute()` method can return false, couldn't the `count()` method do the same ?

Comment (davidyell) [inline src/Query.php]: A count method implies a numerical return, not a boolean.

Comment (ADmad) [inline src/Query.php]: When "something wrong happens" an exception should be thrown :)

**Comment by ADmad** [comment]
Fixed in #44

**Pull request closed without merging**